sympy/parsing/c/c_parser.py

```diff
@@ -330,25 +359,30 @@ def transform_var_decl(self, node):
 
             except StopIteration:
 
-                if (node.type.kind == cin.TypeKind.INT):
-                    type = IntBaseType(String('integer'))
-                    value = Integer(0)
-                elif (node.type.kind == cin.TypeKind.FLOAT):
-                    type = FloatBaseType(String('real'))
-                    value = Float(0.0)
-                elif (node.type.kind == cin.TypeKind.BOOL):
-                    type = Type(String('bool'))
-                    value = false
+                if node.type.kind in self._data_types["int"]:
+                    type = self._data_types["int"][node.type.kind]
+                elif node.type.kind in self._data_types["float"]:
+                    type = self._data_types["float"][node.type.kind]
+                elif node.type.kind in self._data_types["bool"]:
+                    type = self._data_types["bool"][node.type.kind]
                 else:
                     raise NotImplementedError("Only bool, int "
-                            "and float are supported")
+                        "and float are supported")
+                value = None
```

Comment:
no need of that either, just remove value completely from here

Reply:
Alright then

Reply:
Does that look good? I didn't remove it though. :sweat_smile: See whether it is better or not

Reply:
This portion of the code is executed when the variables are declared without a default value like `int a` or `float b`
as the var_decl node will have no children in that case. So it'll be better to just declare the Variable without a 'value'. I think it'll be better to return without a value.

Reply:
I got it, I thought, when there is only declaration, it will be ok to have `None`.  I am changing it now, let me know, if that looks good

Reply:
It is okay to have None, but why waste a new variable and a few extra lines of code when it's going to default to that anyway.